microbench/src/main/java/io/netty/microbench/http2/internal/hpack/HpackUtilBenchmark.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbench.http2.internal.hpack;
+
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.AsciiString;
+import io.netty.util.internal.PlatformDependent;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.List;
+
+@Threads(1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+public class HpackUtilBenchmark extends AbstractMicrobenchmark {
+    @Param
+    public HeadersSize size;
+
+    private List<Header> headers;
+
+    @Setup(Level.Trial)
+    public void setup() {
+        headers = Util.headers(size, false);
+    }
+
+    @Benchmark
+    public int oldEquals() {
+        int count = 0;
+        for (int i = 0; i < headers.size(); ++i) {
+            Header header = headers.get(i);
+            if (oldEquals(header.name, header.name)) {
+                ++count;
+            }
+        }
+        return count;
+    }
+
+    @Benchmark
+    public int newEquals() {
+        int count = 0;
+        for (int i = 0; i < headers.size(); ++i) {
+            Header header = headers.get(i);
+            if (newEquals(header.name, header.name)) {
+                ++count;
+            }
+        }
+        return count;
+    }
+
+    private static boolean oldEquals(CharSequence s1, CharSequence s2) {
```

Comment:
never mind ... I thought this two methods are a copy of what you have implemented...